### ISSUE TITLE
aws-batch-alpha : dependencies fixed to aws-cdk-lib (2.95.1)

### ISSUE THREAD

**Issue by npvisual** [aws/aws-cdk#27699]
### Describe the bug

It appears that the dependencies for aws-batch-alpha require aws-cdk-lib v2.95.1. Since the core CDK libraries are now at v2.103.0 this requirement forces the rest of the dependencies to be stuck at v2.95.1.

### Expected Behavior

Similar to other "alpha" modules, it would be much better to specify the dependency on aws-cdk-lib along the lines of : `>=2.95.1,<3.0.0` so that the rest of the modules can be upgraded to the latest releases.

### Current Behavior

Current behavior (python with poetry) is that the dependency management system will complain because the specified version of aws-cdk-lib is different from the one defined in aws-batch-alpha (which is v2.95.1).

### Reproduction Steps

With python 3.11 and poetry you can specify dependencies in the pyproject.toml as follows : 

```
...
[tool.poetry.dependencies]
python = "^3.11"
"constructs" = "10.2.70"
"aws-cdk-lib" = "2.103.0"
"aws-cdk.aws-batch-alpha" = "2.95.1-alpha.0"
"aws-cdk.aws-apigatewayv2-alpha" = "2.103.0-alpha.0"
"aws-cdk.aws-apigatewayv2-integrations-alpha" = "2.103.0-alpha.0"
"aws-cdk.aws-synthetics-alpha" = "2.101.1-alpha.0"
```

When you run : `poetry install`, it will complain with the following error message : 
> Because aws-cdk-aws-batch-alpha (2.95.1a0) depends on aws-cdk-lib (2.95.1)
 and <app> depends on aws-cdk-lib (2.103.0), aws-cdk-aws-batch-alpha is forbidden.
So, because <app> depends on aws-cdk.aws-batch-alpha (2.95.1-alpha.0), version solving failed.

### Possible Solution

aws-batch-alpha probably needs to define a less restrictive version dependency and use `>=2.95.1,<3.0.0`

### CDK CLI Version

2.103.0 (build d0d7547)

### Framework Version

2.103.0 (well... trying to)

### Node.js Version

v20.6.0

### OS

Debian slim bookworm (12.0)

### Language

Python

### Language Version

Python 3.11

**Comment by npvisual** [comment]
FYI. As a reference, this issue below is a blocker for us (from aws-synthetics-alpha) and was labeled a P1 https://github.com/aws/aws-cdk/issues/27089
It was fixed by PR #27167 but we can't upgrade because of the fixed dependency in aws-batch-alpha
(cc @kaizencc)

**Comment by peterwoodworth** [comment]
That's because batch graduated to stable https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_batch-readme.html
